pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -430,7 +432,25 @@ func buildHTTPFilter(service *serviceMetadata, store model.IstioConfigStore) *ht
 // in service mesh to the corresponding proxy config for the specified service. The generated proxy config
 // will be consumed by envoy RBAC filter to enforce access control on the specified service.
 func convertRbacRulesToFilterConfig(
-	service *serviceMetadata, roles []model.Config, bindings []model.Config) *rbacconfig.RBAC {
+	service *serviceMetadata, roles []model.Config, bindings []model.Config, globalPermissive bool) *rbacconfig.RBAC {
+
+	rbac := &policyproto.RBAC{
+		Action:   policyproto.RBAC_ALLOW,
+		Policies: map[string]*policyproto.Policy{},
+	}
+
+	permissiveRbac := &policyproto.RBAC{
+		Action:   policyproto.RBAC_ALLOW,
+		Policies: map[string]*policyproto.Policy{},
+	}
+
+	// If RBAC Config is set to permissive mode globally, RBAC is transparent to users;
+	// when mapping to rbac filter config, there is only shadow rules(no normal rules).
+	if globalPermissive {
+		return &rbacconfig.RBAC{
+			ShadowRules: permissiveRbac}
```

Comment:
The `permissiveRbac` is always empty, is this expected?

Reply:
I think so, global rbac config in permissive mode indicates rbac isn't enabled just test rbac from 0->1, so it overrides any servicerole/binding.

Reply:
updated to allow set shadow_rules